From ef832a115b40861c08df333339b1366da49e5393 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 17 Apr 2016 12:35:41 -0700
Subject: [PATCH 9/9] WebKitMacros: Append to -I and not to -isystem

gcc-6 has now introduced stdlib.h in libstdc++ for better
compliance and its including the C library stdlib.h using
include_next which is sensitive to order of system header
include paths. Its infact better to not tinker with the
system header include paths at all. Since adding /usr/include
to -system is redundant and compiler knows about it moreover
now with gcc6 it interferes with compiler's functioning
and ends up with compile errors e.g.

/usr/include/c++/6.0.0/cstdlib:75:25: fatal error: stdlib.h: No such file or directory

Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 Source/JavaScriptCore/shell/CMakeLists.txt | 2 +-
 Source/WebCore/PlatformGTK.cmake           | 6 +++---
 Source/WebKit/PlatformGTK.cmake            | 2 +-
 Source/cmake/WebKitMacros.cmake            | 2 +-
 Tools/MiniBrowser/gtk/CMakeLists.txt       | 2 +-
 Tools/TestWebKitAPI/PlatformGTK.cmake      | 2 +-
 6 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/Source/JavaScriptCore/shell/CMakeLists.txt b/Source/JavaScriptCore/shell/CMakeLists.txt
index bc37dd31..4e49871f 100644
--- a/Source/JavaScriptCore/shell/CMakeLists.txt
+++ b/Source/JavaScriptCore/shell/CMakeLists.txt
@@ -35,7 +35,7 @@ WEBKIT_INCLUDE_CONFIG_FILES_IF_EXISTS()
 WEBKIT_WRAP_SOURCELIST(${JSC_SOURCES})
 WEBKIT_WRAP_SOURCELIST(${TESTAPI_SOURCES})
 include_directories(./ ${JavaScriptCore_INCLUDE_DIRECTORIES})
-include_directories(SYSTEM ${JavaScriptCore_SYSTEM_INCLUDE_DIRECTORIES})
+include_directories(${JavaScriptCore_SYSTEM_INCLUDE_DIRECTORIES})
 add_executable(jsc ${JSC_SOURCES})
 target_link_libraries(jsc ${JSC_LIBRARIES})
 
diff --git a/Source/WebCore/PlatformGTK.cmake b/Source/WebCore/PlatformGTK.cmake
index 73506c74..8eb8b415 100644
--- a/Source/WebCore/PlatformGTK.cmake
+++ b/Source/WebCore/PlatformGTK.cmake
@@ -281,7 +281,7 @@ if (ENABLE_PLUGIN_PROCESS_GTK2)
         ${GTK2_INCLUDE_DIRS}
         ${GDK2_INCLUDE_DIRS}
     )
-    target_include_directories(WebCorePlatformGTK2 SYSTEM PRIVATE
+    target_include_directories(WebCorePlatformGTK2 PRIVATE
         ${WebCore_SYSTEM_INCLUDE_DIRECTORIES}
     )
     target_link_libraries(WebCorePlatformGTK2
@@ -305,7 +305,7 @@ add_dependencies(WebCorePlatformGTK WebCore)
 target_include_directories(WebCorePlatformGTK PRIVATE
     ${WebCore_INCLUDE_DIRECTORIES}
 )
-target_include_directories(WebCorePlatformGTK SYSTEM PRIVATE
+target_include_directories(WebCorePlatformGTK PRIVATE
     ${WebCore_SYSTEM_INCLUDE_DIRECTORIES}
     ${GTK_INCLUDE_DIRS}
     ${GDK_INCLUDE_DIRS}
@@ -321,7 +321,7 @@ include_directories(
     "${WEBCORE_DIR}/bindings/gobject/"
 )
 
-include_directories(SYSTEM
+include_directories(
     ${WebCore_SYSTEM_INCLUDE_DIRECTORIES}
 )
 
diff --git a/Source/WebKit/PlatformGTK.cmake b/Source/WebKit/PlatformGTK.cmake
index d83a2e77..401246f4 100644
--- a/Source/WebKit/PlatformGTK.cmake
+++ b/Source/WebKit/PlatformGTK.cmake
@@ -1050,7 +1050,7 @@ if (ENABLE_PLUGIN_PROCESS_GTK2)
     target_include_directories(WebKitPluginProcess2 PRIVATE
         ${WebKit2CommonIncludeDirectories}
     )
-    target_include_directories(WebKitPluginProcess2 SYSTEM PRIVATE
+    target_include_directories(WebKitPluginProcess2 PRIVATE
          ${WebKit2CommonSystemIncludeDirectories}
          ${GTK2_INCLUDE_DIRS}
          ${GDK2_INCLUDE_DIRS}
diff --git a/Source/cmake/WebKitMacros.cmake b/Source/cmake/WebKitMacros.cmake
index 7bc89543..d9818fa4 100644
--- a/Source/cmake/WebKitMacros.cmake
+++ b/Source/cmake/WebKitMacros.cmake
@@ -78,7 +78,7 @@ macro(WEBKIT_FRAMEWORK_DECLARE _target)
 endmacro()
 
 macro(WEBKIT_FRAMEWORK _target)
-    include_directories(SYSTEM ${${_target}_SYSTEM_INCLUDE_DIRECTORIES})
+    include_directories(${${_target}_SYSTEM_INCLUDE_DIRECTORIES})
     target_sources(${_target} PRIVATE
         ${${_target}_HEADERS}
         ${${_target}_SOURCES}
diff --git a/Tools/MiniBrowser/gtk/CMakeLists.txt b/Tools/MiniBrowser/gtk/CMakeLists.txt
index e832a86d..ce92c864 100644
--- a/Tools/MiniBrowser/gtk/CMakeLists.txt
+++ b/Tools/MiniBrowser/gtk/CMakeLists.txt
@@ -57,7 +57,7 @@ endif ()
 add_definitions(-DGDK_VERSION_MIN_REQUIRED=GDK_VERSION_3_6)
 
 include_directories(${MiniBrowser_INCLUDE_DIRECTORIES})
-include_directories(SYSTEM ${MiniBrowser_SYSTEM_INCLUDE_DIRECTORIES})
+include_directories(${MiniBrowser_SYSTEM_INCLUDE_DIRECTORIES})
 add_executable(MiniBrowser ${MiniBrowser_SOURCES})
 target_link_libraries(MiniBrowser ${MiniBrowser_LIBRARIES})
 
diff --git a/Tools/TestWebKitAPI/PlatformGTK.cmake b/Tools/TestWebKitAPI/PlatformGTK.cmake
index 1be3dd52..7bdddf37 100644
--- a/Tools/TestWebKitAPI/PlatformGTK.cmake
+++ b/Tools/TestWebKitAPI/PlatformGTK.cmake
@@ -20,7 +20,7 @@ include_directories(
     ${WEBKIT2_DIR}/UIProcess/API/gtk
 )
 
-include_directories(SYSTEM
+include_directories(
     ${GDK3_INCLUDE_DIRS}
     ${GLIB_INCLUDE_DIRS}
     ${GTK3_INCLUDE_DIRS}
-- 
2.14.1

